compiler/noirc_frontend/src/hir_def/types.rs

```diff
@@ -983,6 +983,34 @@ impl Type {
         }
     }
 
+    pub fn is_primitive(&self) -> bool {
+        match self.follow_bindings() {
+            Type::FieldElement
+            | Type::Array(_, _)
+            | Type::Slice(_)
+            | Type::Integer(..)
+            | Type::Bool
+            | Type::String(_)
+            | Type::FmtString(_, _)
+            | Type::Unit
+            | Type::Function(..)
+            | Type::Tuple(..) => true,
+            Type::Alias(alias_type, generics) => {
+                alias_type.borrow().get_type(&generics).is_primitive()
+            }
+            Type::MutableReference(typ) => typ.is_primitive(),
+            Type::Struct(..)
+            | Type::TypeVariable(..)
+            | Type::TraitAsType(..)
+            | Type::NamedGeneric(..)
```

Comment:
type variables & named generics can be primitive if they're bound to one

Reply:
I see. I was about to fix this but I noticed there's a `follow_bindings()` call at the beginning. Would that take care of this?

Reply:
Ah, I missed that as well. It's probably fine then